### DISCUSSION TITLE
Using "Proxy mode" to a API on AWS - but via our corporate proxy - possible?

### DISCUSSION THREAD

**Discussion by brendanheyu** [mockoon/mockoon#685] [Help] [answered]
Hi folks!

I'm trying to consume an API we have on AWS. My dev site will end up residing in an S3 bucket that shares the domain and hence averts any CORs issues. When developing locally however, I would like to use that same API, but but make calls from my localhost webserver. CORs rules that out directly.

I was hoping Mockoon would help me proxy my API calls - and negotiate my corporate proxy (that asks for user/pass) at the same time. [localhost > mockoon > corp proxy > AWS API]

At this point I think I'm stuck - hopefully I just don't know quite enough?

**Comment by 255kb** [comment]
Hi @brendanheyu 
You should be able to proxy to a server using basic auth by passing an `Authorization: Basic xxxx` header in the proxy request headers. These headers are automatically added to the request when forwarded to the proxied server:
It could look like this:
![image](https://user-images.githubusercontent.com/7489814/163527123-743c5974-01a6-4124-8eff-b6081bd6305d.png)
Replace "admin" and "supersecret" by your proxy basic auth credentials: `Basic {{# base64}}admin:supersecret{{/base64}}` 

I hope this helps.

**Reply by brendanheyu** [reply]
Hi @255kb ,

Thanks for taking the time to respond :)

I've attempted what you have explained above. I still get the same issue, so I might ask/explain some more just to see if I have the correct understanding.

I have my current values (with an actual URL, user and pass) as such:

![image](https://user-images.githubusercontent.com/1276819/164133590-b76f7dc5-54a3-4f7b-bf93-df3a8667b4ba.png)

* There is no proxy setting in the Mockoon app, so I assume it obtains my corp proxy address from the Windows settings for my corp proxy server? 
* We have multiple proxies available, one for each branch location (Australia is a large place!). When I change my windows info to use another branch's proxy - the IP address in the ECONNREFUSED message does not change... I'm suspicious! Thoughts on this?

* The error appears as a 504? 
![image](https://user-images.githubusercontent.com/1276819/164133868-695cbdab-ae45-48be-b532-725add610ea4.png)

I can almost smell the fresh API smell just on the other side of this error - so close!

Thanks for looking,

Brendan

**Reply by 255kb** [reply, marked as answer]
Sorry for the late reply. 
I investigated a bit and as we are using Node.js' HTTP library, Mockoon is *not* using your operating system proxy settings automatically. 
I am afraid I have no workaround for now to force Mockoon to use the system's proxy settings. 

I suggest that you open an issue so we can track this feature request.

**Reply by brendanheyu** [reply]
Sorry for _my_ late reply!

Thanks for having a look - I wasn't too sure if I was missing something! If you think this feature is worthwhile I will raise a request - let me know!
